### DISCUSSION TITLE
Notification custom mail server - works only if the server owns a CA released SSL/TLS Certificate

### DISCUSSION THREAD

**Discussion by larrykind** [globaleaks/globaleaks-whistleblowing-software#4544] [General]
**Current behavior**

In "Notification settings", if a custom mailserver is set  who doesn't own a CA released SSL/TLS certificate, notification emails are not sent at all. This added security feature is not requested in the settings.

**Expected behavior**

If a custom mailserver is set in the Notification settings, there shold be a flag to tell globaleaks that the mail server owns a CA released SSL/TLS certificate.
It's a good security feature, but I think It should be optional, and the user should be aware of this.

**Steps to reproduce the problem or feature illustration**

1. Login as admin 
2. In notification settings set a mailserver that doesn't own a CA signed SSL/TLS certificate, **or set a self-signed one**
3. Try to do whatever operation that sends an email (ticket, password change, etc.): mails are not sent.

**What is the motivation or use case for changing the behavior?**

To make the notification emails work for every settings with a little lack of security also, for all kind of users.

**GlobaLeaks version:**

4.0.18

**Browser:**

Firefox 75, Chrome 88

**Server Operating System and Version (if applicable):**
 
debian 10.3 buster

**Client Operating System and Version (if applicable):**

windows 10

**Comment by evilaliv3** [comment]
Thank you for the feedback @larrykind.

actually we consider that this feature is fundamental. We may raise awareness via the documentation but by the current analysis we would prefer to not open to the possibility that users could disable the certificate chain validation.

Alternatively i would rather make it possible to load manually the full chain marking it as trustworty applying a TOFU policy.

What do you think?

**Comment by fpietrosanti** [comment]
@evilaliv3 what if we enable to load a certificate store (being a single public key or a CA certificates chain) that override the operating system one? 

That way it could be possible to handle all conditions, from the internal corporate CA of a Microsoft Exchange to send email up to a self-signed

**Comment by larrykind** [comment]
Hi @evilaliv3 and @fpietrosanti , 
in my opinion the fundamental is make the whole chain to work without too much hang for the administrator.
Since PGP key for the Users is not mandatory ( that should be much more important that notification email Certificate ), for the same reason a TOFU policy for notification emails, that are not containing sensitive whistleblower data, should be optional.

**Comment by evilaliv3** [comment]
Thank you for your feedback @larrykind.

Your feedback is complete correct in relation to notifications about submission; my concern in on emails sent for issuing password resets.

@fpietrosanti: Yes this was actually the idea i described; we could at global platform level make it possible to add a set of certificates to be added to the certificate store. More than an override i would make it possible to append certificates not included in the os store.

**Comment by fpietrosanti** [comment]
@evilaliv3 If we enable only to "add" we would providing greater compatibility to fix this kind of problem without dealing with a command line, but it would not become a "security" feature (i.e. if we restrict ca store for validation to a single public certificate or ca-root, we'd be increasing security), let's consider while implementing it whenever making it a configurable behaviour (add to existing or override existing)

**Comment by evilaliv3** [comment]
I see. That would actually work. In this case we would need to override the certificate store just for the SMTP.

In my feature i was instead thining to enable to load appen to the global certificate store to patch other missing certificates for other situations.

As this is not that urgent at the moment, let think about this for a while evaluating the other needed we may have.

**Comment by larrykind** [comment]
Hi, 
To make globaleaks ready to go after install, I think it should be followed a best effort policy with encryption mechanisms. Basically, I think it shoud completely work everytime, by making the tenants administrators (and maybe the email receivers) aware that there is a security downgrade.

**Comment by evilaliv3** [comment]
I definitely agree.

The application should start to include some tutorial as wel (that is planned as topic for next UI/UX updates).

@larrykind: as this topic is quite different from what originally discussed do you mind to check if other existing tickets  are already dealing with this (i think so actually) or open a new one?

**Comment by larrykind** [comment]
Hi @evilaliv3, 
referring to this issue, I thought that the best effort policy we talked about could be achieved through using certificates for the mailserver only if available, giving a "pass" advertisement to the tenant administrator. If certificates (for the mail server) are not used or expired (or errors, etc,..), a warning message should be given to the adiminstrator, but email service should continue to work.

If there will be available a mechanism as you and @fpietrosanti described, in the way that the administrator could easily set up a certificate just by clicking some backend button (as well done for LE certificates for HTTPS), it will be great. But It should be considered that not every administrator will ever do this procedure in setup phase, being sure that the system will work fine in http only / emails without encryption.

**Comment by evilaliv3** [comment]
I get your frustration in a system that would not work by default or easily
but actually we consider that a system witout https and without a password
reset mechanism protected by a valid certificat would result too much in a
compromised system. This has been the case of many important systems like
ThaiTobacco (http://ept.thaitobacco.or.th/) that runs in http and
irresponsibly don't answer any attempt to advise and suggest them.

For this reason for example right know you find that submission can happen
only after that recipients have performed first login that is what required
for generating encryption keys

I would be happy if you could continue to provide suggestions and try to
understand with us how we can better cope with this issues. Thank you!

**Comment by mttstt** [comment]
So now I can't use anymore smtp.gmail.com ?

**Comment by mttstt** [comment]
Version 4.0,32
With the "mail.globaleaks.org" server the "Test of configuration" works correctly, instead the notification mails (ticket, password change, etc.) are not sent.

**Comment by mttstt** [comment]
Solved, restarting the server!
